instrumentation/spymemcached-2.12/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spymemcached/SpymemcachedRequest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spymemcached;
+
+import com.google.auto.value.AutoValue;
+import net.spy.memcached.MemcachedConnection;
+
+@AutoValue
+public abstract class SpymemcachedRequest {
+
+  public static SpymemcachedRequest create(MemcachedConnection connection, String statement) {
+    return new AutoValue_SpymemcachedRequest(connection, statement);
+  }
+
+  public abstract MemcachedConnection getConnection();
+
+  public abstract String getStatement();
+
+  public String dbOperation() {
+    char[] chars =
+        getStatement()
+            .replaceFirst("^async", "")
```

Comment:
These look like they can have significant performance overhead on all statements, not just these two, because they compile the regex every single time. How about `startsWith` + `substring`?

Reply:
I realize this is copy-pasted from the old code - please file an issue if don't want to fix in this PR. Very obvious when reviewing it though :)